src/triangle.rs

```diff
@@ -0,0 +1,651 @@
+
+    fn next(&mut self) -> Option<PathEl> {
+        self.ix += 1;
+        match self.ix {
+            1 => Some(PathEl::MoveTo(self.triangle.a)),
+            2 => Some(PathEl::LineTo(self.triangle.b)),
+            3 => Some(PathEl::LineTo(self.triangle.c)),
+            4 => Some(PathEl::ClosePath),
+            _ => None,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{Point, Size, Triangle};
+
+    #[test]
+    fn from_centroid_sizes() {
+        let test = Triangle::from_centroid_distances((0.06, 12.2), [11.9, 1.2, 12.3]);
+        let expected = Triangle::from_coords(
+            (0.06, 24.1),
+            (-0.7885281374238573, 11.351471862576142),
+            (8.757413408594534, 3.5025865914054624),
+        );
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn from_centroid_distances() {
+        let test = Triangle::from_centroid_distances((0.6, 12.2), [11.9, 1.2, 12.3]);
+        let expected = Triangle::from_coords(
+            (0.6, 24.1),
+            (-0.24852813742385726, 11.351471862576142),
+            (9.297413408594533, 3.5025865914054624),
+        );
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn from_radius() {
+        let test = Triangle::from_radius(10.1);
+        let expected = Triangle::from_coords(
+            (0.0, 20.199999999999996),
+            (-17.493713156445658, -10.099999999999998),
+            (17.493713156445658, -10.099999999999998),
+        );
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn with_centroid() {
+        let test = Triangle::EQUILATERAL.with_centroid((12.2, -2.3));
+        let expected = Triangle::from_coords((12.2, 1.7000000000000002), (9.2, -4.3), (15.2, -4.3));
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn centroid() {
+        let test = Triangle::from_coords((-90.02, 3.5), (7.2, -9.3), (8.0, 9.1)).centroid();
+        let expected = Point::new(-24.939999999999998, 1.0999999999999996);
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn sizes() {
+        let test = Triangle::from_coords((-20.0, 180.2), (1.2, 0.0), (290.0, 100.0)).sizes();
+        let expected = [
+            Size::new(-110.39999999999999, 86.8),
+            Size::new(-89.19999999999999, -93.39999999999999),
+            Size::new(199.60000000000002, 6.6000000000000085),
+        ];
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn perimeter() {
+        let test = Triangle::new(
+            (781239.273894, 789234.234789),
+            (234897.823471, 378902.234789),
+            (789241.789234, 789234.234897),
+        )
+        .perimeter();
+        let expected = 1380963.0638877784;
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn area() {
+        let test = Triangle::new(
+            (12123.423, 2382.7834),
+            (7892.729, 238.459),
+            (7820.2, 712.23),
+        )
+        .area();
+        let expected = 1079952.91574081;
```

Comment:
dumb floating point nonsense. tests dont pass because of this, how should we handle it? other places use `assert_approx_eq()`